eZ/Bundle/EzPublishCoreBundle/Imagine/AliasGenerator.php

```diff
@@ -63,18 +64,25 @@ class AliasGenerator implements VariationHandler
      */
     private $ioResolver;
 
+    /**
+     * @var IOServiceInterface
+     */
+    private $ioService;
+
     public function __construct(
         LoaderInterface $dataLoader,
         FilterManager $filterManager,
         ResolverInterface $ioResolver,
         FilterConfiguration $filterConfiguration,
-        LoggerInterface $logger = null
+        LoggerInterface $logger = null,
+        IOServiceInterface $ioService
```

Comment:
This also has to be optional, since `$logger` is. Method needs to return `null` for `with` and `height` if service is not present. Additionally I'd trigger here in the constructor `E_USER_WARNING` with information that this is missing.

Reply:
> We don't promise BC on constructors, right?

It depends :)
If you're decorating and calling parent constructor, this would fail.
Of course if you're decorating this one, you're doing it at your own risk.
But the problem is that for bugfix releases such things shouldn't change, so rather make it optional with a warning.

Reply:
As much as possible, we should not break BC. Including on constructors. In that case, make it optional, and don't use it if not set. I think it's simple enough. Do you guys agree ?

My main concern would be to track these so that we can clean them up when we have a chance.

Reply:
Thanks, guys. I've made that dependency optional and trigger silenced error with a warning message. Done in: https://github.com/ezsystems/ezpublish-kernel/pull/2310/commits/39d3fed005025bbba1480d41e6d9676572039f7c